snakemake/common.rules

```diff
@@ -28,27 +30,43 @@ _READ_LENGTH = 124
 _STAR_GENOME_DIR = os.path.join(
     os.path.dirname(config["reference"]["genome"]), "star-genome-%d" % _READ_LENGTH)
 
+
 # Common functions
 
 def _get_half_cores(_):
-  return max(1, int(config["num_threads"]/2))
+  n = _get_all_cores(_)
+  return max(1, n // 2)
 
 def _get_all_cores(_):
-  return int(config["num_threads"])
+  if "num_threads" in config:
+    return int(config["num_threads"])
+  else:
+    return psutil.cpu_count()
 
-def _get_threads_for_alignment(_):
-  if _mem_gb_for_alignment() < 20:
-    return _get_all_cores(_)
+def _available_memory_gb():
+  if "mem_gb" in config:
+    return int(config["mem_gb"])
   else:
-    return _get_half_cores(_)
+    bytes_per_gb = 1024 ** 3
+    return max(1, psutil.virtual_memory().total // bytes_per_gb)
```

Comment:
it seems weird to have default resource logic both here and in run_snakemake.py - i don't have strong feelings about where it should live, happy to have it here as long as we remove it from run_snakemake.py